### PR TITLE
[feature] new resources: views, view grants and database grants

### PR DESCRIPTION
**Pull request by sjauld**
Apologies in advance for this one.

This PR covers the addition of views (#36), view grants and database grants (#26). It's a bit of a beast. There is also a small amount of refactoring. It currently also includes the changes from #39 which is yet to be merged.

I have got acceptance testing on [our fork in Travis](https://travis-ci.com/Viostream/terraform-provider-snowflake/builds/117067677), to achieve this I have to skip the managed account acceptance test since my test account doesn't yet have the ability to create managed accounts.

Please let me know how you want to proceed with this one - happy to split it up into 3x smaller PRs if that will help.

### CODE REVIEW

File: pkg/resources/grant_helpers.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/jmoiron/sqlx"
+
+	"github.com/chanzuckerberg/terraform-provider-snowflake/pkg/snowflake"
+)
+
+// grant represents a generic grant of a privilge from a grant (the target) to a
+// grantee. This type can be used in conjunction with github.com/jmoiron/sqlx to
+// build a nice go replresentation of a grant
```
Comment (edulop91) [inline pkg/resources/grant_helpers.go]:
```suggestion
// build a nice go representation of a grant
```

**Comment by ryanking** [comment]
@sjauld can you resolve the merge conflict here?

**Comment by ryanking** [comment]
@sjauld is this duplicated by https://github.com/chanzuckerberg/terraform-provider-snowflake/pull/52 ?

**Comment by sjauld** [comment]
Yes I think everything here is also in #52. Happy for this one to be
closed.

**Pull request closed without merging**